src/Form/AuthenticationForm.php

```diff
@@ -167,13 +134,27 @@ protected function getEditableConfigNames() {
 
   /**
    * {@inheritdoc}
-   *
-   * @throws \Drupal\Core\Entity\EntityStorageException
-   * @throws \Drupal\key\Exception\KeyValueNotSetException
    */
   public function buildForm(array $form, FormStateInterface $form_state) {
-    $form = parent::buildForm($form, $form_state);
+    // Save the key for later use.
+    $this->activeKey = $this->getActiveKey();
 
+    /** @var \Drupal\apigee_edge\Plugin\KeyProviderRequirementsInterface $key_provider */
+    if (($key_provider = $this->activeKey->getKeyProvider()) instanceof KeyProviderRequirementsInterface) {
+      try {
+        $key_provider->checkRequirements($this->activeKey);
+      }
+      catch (KeyProviderRequirementsException $exception) {
+        $this->messenger()->addError($this->t("The requirements of the selected key provider (@key_provider) are not fulfilled. Fix the errors described below or <a href=':key_config_uri' target='_blank'>change the active key's provider</a>.", [
+          '@key_provider' => $this->activeKey->getKeyProvider()->getPluginDefinition()['label'],
+          ':key_config_uri' => Url::fromRoute('entity.key.edit_form', ['key' => $this->activeKey->id()])->toString(),
+        ]));
+        $this->messenger()->addError($exception->getTranslatableMarkupMessage());
```

Comment:
`getTranslatableMarkupMessage()` can return null, you should check whether the return value is null and in that case return a "Something went wrong..." message, in that case, you could display the non-translateable exception message maybe.